rust/connlib/clients/headless/src/main.rs

```diff
@@ -72,7 +73,12 @@ fn main() -> Result<()> {
     let secret = parse_env_var::<String>(SECRET_ENV_VAR)?;
     let mut session = Session::connect(url, secret, CallbackHandler).unwrap();
     tracing::info!("Started new session");
-    session.wait_for_ctrl_c().unwrap();
+
+    let (tx, rx) = std::sync::mpsc::channel();
+    ctrlc::set_handler(move || tx.send(()).expect("Could not send stop signal on channel."))
+        .expect("Error setting Ctrl-C handler");
+    rx.recv().expect("Could not receive ctrl-c signal");
```

Comment:
But that can only happen with a logic error since `tx` should only be dropped after `rx` is signaled , I'd like to detect it as a panic.

Reply:
It's not very important anyways, since the headless client as it stands will only be used for testing in the released version it'll probably run as a service.